lib/web_ui/lib/src/engine/window.dart

```diff
@@ -760,6 +800,13 @@ class EngineWindow extends ui.Window {
     });
   }
 
+  void _addUrlStrategyListener() {
+    _onUrlStrategy = allowInterop(setUrlStrategy);
+    registerHotRestartListener(() {
```

Comment:
Another benefit of the top-level `window.flutter.` namespace would be that we can easily uninitialize it in one place (just set the whole thing to null). But OK with leaving this as a future exercise.